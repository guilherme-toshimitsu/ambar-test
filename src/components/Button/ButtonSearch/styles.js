const styles = theme => ({
  icon: {
    width: 24,
    height: 24,
  },
  button: {
    width: 76,
    height: 52,
    boxShadow: 'none',
  },
});

export default styles;
