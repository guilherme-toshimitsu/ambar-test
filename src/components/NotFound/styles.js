const styles = {
  container: {
    width: '100%',
    // height: '100%',
    minHeight: 317,
    backgroundColor: '#ffffff',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconBg: {
    width: 116,
    height: 116,
    backgroundColor: '#0086ff',
    backgroundColor: 'rgba(0, 134, 255, 0.2)',
    borderRadius: 50,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 'auto',
  },
  icon: {
    width: 40,
    height: 36,
    color: '#0086ff',
  },
  text: {
    width: 307,
    height: 19,
    fontSize: 14,
    color: '#404040',
  },
};
export default styles;
