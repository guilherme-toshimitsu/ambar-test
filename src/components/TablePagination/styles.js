const styles = theme => ({
  buttonsContainer: {
    display: 'flex',
    width: '100%',
  },
  previous: {
    textAlign: 'right',
    width: '46%',
  },
  current: {
    textAlign: 'center',
    width: '8%',
  },
  next: {
    textAlign: 'left',
    width: '46%',
  },
});

export default styles;
