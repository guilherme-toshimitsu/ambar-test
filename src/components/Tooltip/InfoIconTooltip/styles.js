const styles = theme => ({
  tooltip: {
    width: 220,
    height: 96,
    backgroundColor: '#404040',
    fontFamily: 'OpenSans',
    color: '#ffffff',
    paddingTop: 12,
  },
  arrow: {
    color: '#404040',
  },
  infoIcon: {
    width: 16,
    height: 16,
    color: '#8c8c8c',
    marginBottom: -3,
    marginLeft: 5,
  },
});

export default styles;
