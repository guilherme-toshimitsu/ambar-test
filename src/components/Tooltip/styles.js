const styles = {
  tooltip: {
    fontSize: 12,
    letterSpacing: 0.2,
    wordWrap: 'break-word',
  },
  arrow: {},
};

export default styles;
