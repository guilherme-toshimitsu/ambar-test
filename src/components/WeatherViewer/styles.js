const styles = theme => ({
  paper: {
    height: 400,
    width: 600,
    margin: 'auto',
    padding: '20px 20px 20px 20px',
  },
  loadingRoot: {
    height: '100%',
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  cardContainer: {
    height: '100%',
    width: '100%',
    display: 'flex',
  },
  cardData: {
    width: '100%',
  },
  actionContainer: {
    display: 'flex',
    width: '100%',
    height: 250,
  },
  iconContainer: {
    display: 'flex',
    width: '50%',
    height: 250,
    alignItems: 'flex-end',
  },
  icon: {
    height: 80,
    width: 80,
  },
  buttonContainer: {
    width: '50%',
    height: 250,
    display: 'flex',
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
  },
});

export default styles;
