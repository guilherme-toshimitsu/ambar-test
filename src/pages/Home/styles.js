const styles = {
  buttonsContainer: {
    marginTop: 50,
    marginBottom: 50,
    display: 'flex',
    justifyContent: 'center',
  },
  button: {
    marginRight: 10,
    marginLeft: 10,
  },
};

export default styles;
