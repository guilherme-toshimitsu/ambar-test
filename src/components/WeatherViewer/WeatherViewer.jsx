import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';
import { Paper, CircularProgress, Typography, Button } from '@material-ui/core';
import ControlledLoading from '@components/ControlledLoading';

import styles from './styles';

const useStyles = makeStyles(styles, { name: 'WeatherViewer' });

const hasWeatherIcon = weather => !!(weather.length && weather[0].icon);

const treatIcon = icon => {
  if (icon.includes('.png')) {
    return icon.split('.png')[0] + '.png';
  }

  if (icon.includes('.jpg')) {
    return icon.split('.jpg')[0] + '.jpg';
  }

  return false;
};

const WeatherViewer = ({ main, weather, isLoading, name, history }) => {
  const classes = useStyles();

  return (
    <Paper elevation={3} className={classes.paper}>
      <ControlledLoading
        isLoading={isLoading}
        loadingComponent={
          <div className={classes.loadingRoot}>
            <CircularProgress />
          </div>
        }
        renderComponent={
          <div className={classes.cardContainer}>
            {Object.keys(main).length ? (
              <div className={classes.cardData}>
                <Typography>{`Local: ${name}`}</Typography>
                <Typography>{`Temperatura: ${main.temp} ° Celsius`}</Typography>
                <Typography>{`Temperatura Minima: ${main.temp_min} ° Celsius`}</Typography>
                <Typography>{`Temperatura Máxima: ${main.temp_max} ° Celsius`}</Typography>
                <div className={classes.actionContainer}>
                  <div className={classes.iconContainer}>
                    {hasWeatherIcon(weather) ? (
                      <img src={treatIcon(weather[0].icon)} className={classes.icon} alt={'icone'} />
                    ) : (
                      <Typography>{'Não possui Ícone'}</Typography>
                    )}
                  </div>
                  <div className={classes.buttonContainer}>
                    <Button onClick={() => history.push('/temperature')} color="primary" variant="contained">
                      {`Ir Para: ${name}`}
                    </Button>
                  </div>
                </div>
              </div>
            ) : (
              <>Clique em um Botão escolher a cidade</>
            )}
          </div>
        }
      />
    </Paper>
  );
};

WeatherViewer.defaultProps = {
  main: {},
  weather: [],
  isLoading: false,
  name: '',
};

WeatherViewer.propTypes = {
  main: PropTypes.object,
  weather: PropTypes.array,
  isLoading: PropTypes.bool,
  name: PropTypes.string,
  history: PropTypes.object.isRequired,
};

export default WeatherViewer;
