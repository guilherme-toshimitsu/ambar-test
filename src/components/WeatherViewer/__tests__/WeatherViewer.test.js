import React from 'react';
import { ThemeProvider } from '@material-ui/styles';
import { render, fireEvent } from '@testing-library/react';
import { createMuiTheme } from '@material-ui/core';
import WeatherViewer from '../';

describe('<WeatherViewer> Route Component Loader', () => {
  it('renders default loading', () => {
    const component = render(<WeatherViewer />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders without icon', () => {
    const component = render(<WeatherViewer main={{ temp_min: 0, temp_max: 99 }} weather={[]} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders with icon but not parsable', () => {
    const component = render(<WeatherViewer main={{ temp_min: 0, temp_max: 99 }} weather={[{ icon: 'bla' }]} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders with icon png', () => {
    const component = render(<WeatherViewer main={{ temp_min: 0, temp_max: 99 }} weather={[{ icon: 'bla.png' }]} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders with icon jpg', () => {
    const component = render(<WeatherViewer main={{ temp_min: 0, temp_max: 99 }} weather={[{ icon: 'bla.jpg' }]} />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders and clickButton', () => {
    const component = render(
      <ThemeProvider theme={createMuiTheme({})}>
        <WeatherViewer
          main={{ temp_min: 0, temp_max: 99 }}
          weather={[{ icon: 'bla.jpg' }]}
          history={{ push: () => {} }}
        />
      </ThemeProvider>
    );
    const button = component.getByTestId('go-to-temperature');
    fireEvent.click(button);
  });
});
