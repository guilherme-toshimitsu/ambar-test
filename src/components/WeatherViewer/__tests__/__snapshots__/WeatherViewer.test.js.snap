// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherViewer> Route Component Loader renders default loading 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-1 MuiPaper-rounded"
  >
    <div
      class="WeatherViewer-cardContainer-3"
    >
      Clique em um Botão escolher a cidade
    </div>
  </div>
</DocumentFragment>
`;

exports[`<WeatherViewer> Route Component Loader renders with icon but not parsable 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-142 MuiPaper-rounded"
  >
    <div
      class="WeatherViewer-cardContainer-144"
    >
      <div
        class="WeatherViewer-cardData-145"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Local: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura: undefined ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Minima: 0 ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Máxima: 99 ° Celsius
        </p>
        <div
          class="WeatherViewer-actionContainer-146"
        >
          <div
            class="WeatherViewer-iconContainer-147"
          >
            <img
              alt="icone"
              class="WeatherViewer-icon-148"
            />
          </div>
          <div
            class="WeatherViewer-buttonContainer-149"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="go-to-temperature"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Ir Para: 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<WeatherViewer> Route Component Loader renders with icon jpg 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-354 MuiPaper-rounded"
  >
    <div
      class="WeatherViewer-cardContainer-356"
    >
      <div
        class="WeatherViewer-cardData-357"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Local: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura: undefined ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Minima: 0 ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Máxima: 99 ° Celsius
        </p>
        <div
          class="WeatherViewer-actionContainer-358"
        >
          <div
            class="WeatherViewer-iconContainer-359"
          >
            <img
              alt="icone"
              class="WeatherViewer-icon-360"
              src="bla.jpg"
            />
          </div>
          <div
            class="WeatherViewer-buttonContainer-361"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="go-to-temperature"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Ir Para: 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<WeatherViewer> Route Component Loader renders with icon png 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-248 MuiPaper-rounded"
  >
    <div
      class="WeatherViewer-cardContainer-250"
    >
      <div
        class="WeatherViewer-cardData-251"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Local: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura: undefined ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Minima: 0 ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Máxima: 99 ° Celsius
        </p>
        <div
          class="WeatherViewer-actionContainer-252"
        >
          <div
            class="WeatherViewer-iconContainer-253"
          >
            <img
              alt="icone"
              class="WeatherViewer-icon-254"
              src="bla.png"
            />
          </div>
          <div
            class="WeatherViewer-buttonContainer-255"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="go-to-temperature"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Ir Para: 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<WeatherViewer> Route Component Loader renders without icon 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-36 MuiPaper-rounded"
  >
    <div
      class="WeatherViewer-cardContainer-38"
    >
      <div
        class="WeatherViewer-cardData-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Local: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura: undefined ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Minima: 0 ° Celsius
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Temperatura Máxima: 99 ° Celsius
        </p>
        <div
          class="WeatherViewer-actionContainer-40"
        >
          <div
            class="WeatherViewer-iconContainer-41"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Não possui Ícone
            </p>
          </div>
          <div
            class="WeatherViewer-buttonContainer-43"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              data-testid="go-to-temperature"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Ir Para: 
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
